shared/speaker_setups.hpp

```diff
@@ -0,0 +1,441 @@
+      Speaker{"AC_00010006", "Rs", "M-110", -110.f, 0.f, Layer::middle, false, false},
+      Speaker{"AC_0001000a", "Lss", "M+090", 90.f, 0.f, Layer::middle, false, false},
+      Speaker{"AC_0001000b", "Rss", "M-090", -90.f, 0.f, Layer::middle, false, false},
+      Speaker{"AC_0001001a", "BL", "M+135", 135.f, 0.f, Layer::middle, false, false}, // This speaker is marked as diffuse
+      Speaker{"AC_0001001b", "BR", "M-135", -135.f, 0.f, Layer::middle, false, false},  // This speaker is marked as diffuse
+      Speaker{"AC_0001000d", "Ltf", "U+030", 30.f, 45.f, Layer::upper, false, false},
+      Speaker{"AC_0001000f", "Rtf", "U-030", -30.f, 45.f, Layer::upper, false, false},
+      Speaker{"AC_0001000e", "TpFC", "U+000", 0.f, 45.f, Layer::upper, false, false},
+      Speaker{"AC_00010010", "Ltr", "U+110", 110.f, 45.f, Layer::upper, false, false},
+      Speaker{"AC_00010012", "Rtr", "U-110", -110.f, 45.f, Layer::upper, false, false},
+      Speaker{"AC_00010013", "TpSiL", "U+090", 90.f, 45.f, Layer::upper, false, false},
+      Speaker{"AC_00010014", "TpSiR", "U-090", -90.f, 45.f, Layer::upper, false, false},
+      Speaker{"AC_0001001e", "Ltb", "U+135", 135.f, 30.f, Layer::upper, false, false},
+      Speaker{"AC_0001001f", "Rtb", "U-135", -135.f, 30.f, Layer::upper, false,false}
+    }),
+
+  /* clang-format on */
+};
+
+inline SpeakerSetup speakerSetupByName(const std::string& name) {
+  auto it = std::find_if(
+      SPEAKER_SETUPS.begin(), SPEAKER_SETUPS.end(),
+      [name](auto speakerSetup) { return speakerSetup.name == name; });
+  if (it != SPEAKER_SETUPS.end()) {
+    return *it;
+  } else {
+    return SpeakerSetup{};
+  }
+}
+
+inline SpeakerSetup speakerSetupByCommonName(const std::string& commonName) {
+  auto it = std::find_if(SPEAKER_SETUPS.begin(), SPEAKER_SETUPS.end(),
+                         [commonName](auto speakerSetup) {
+                           return speakerSetup.commonName == commonName;
+                         });
+  if (it != SPEAKER_SETUPS.end()) {
+    return *it;
+  } else {
+    return SpeakerSetup{};
+  }
+}
+
+inline SpeakerSetup speakerSetupByIndex(int index) {
+  if (0 <= index && index < SPEAKER_SETUPS.size()) {
+    return SPEAKER_SETUPS.at(index);
+  } else {
+    return SpeakerSetup{};
+  }
+}
+
+inline SpeakerSetup speakerSetupByPackFormatId(const std::string& packFormatId) {
+  auto it = std::find_if(
+    SPEAKER_SETUPS.begin(), SPEAKER_SETUPS.end(),
+    [packFormatId](auto speakerSetup) { return speakerSetup.packFormatId == packFormatId; });
+  if (it != SPEAKER_SETUPS.end()) {
+    return *it;
+  }
+  else {
+    return SpeakerSetup{};
+  }
+}
+
+inline int getIndexFromLegacySpeakerSetupIndex(int legacyIndex) {
+  for(int i = 0; i < SPEAKER_SETUPS.size(); i++) {
+    if(SPEAKER_SETUPS[i].legacySpeakerSetupIndex.has_value()) {
+      if(SPEAKER_SETUPS[i].legacySpeakerSetupIndex.value() == legacyIndex) {
+        return i;
+      }
+    }
+  }
+  return -1;
+}
```

Comment:
`std::optional` has implicit conversion to bool for checking if it contains a value, and the dereference operator for getting the contents (mimicing pointer interface), so I'd do the following:
```c++
// std::size_t so no sign mismatch on comparison, you can do auto i = 0ul but it's iffy on some platforms  
// prefix increment so no temp (will get optimized out probably but habit) 
for(std::size_t i = 0; i != SPEAKER_SETUPS.size(); ++i) { 
  auto candidateIndex = SPEAKER_SETUPS[i].legacySpeakerSetupIndex;
  if(candidateIndex && *candidateIndex == legacyIndex) {
    return i;
  } 
}
```